catkit2/services/phasics_cam/phasics_cam.py

```diff
@@ -0,0 +1,286 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Oct 10 16:56:37 2024
+
+@author: lameier - using original code by Hari who worked with Phasics
+"""
+import PySID4x as sid
+import numpy as np
+from catkit2.testbed.service import Service
+import threading
+
+class IllegalArgumentError(ValueError):
+    pass
+
+class PhasicsCam(Service):
+    """
+    A service class for the Phasics camera.
+
+    A camera service with functionality for taking measurements,
+    starting and stopping live acquisitions, and managing an image buffer.
+
+    Parameters
+    ----------
+    Service : catkit2.testbed.service.Service
+        The base service class from which this class inherits.
+    """
+    def __init__(self):
+        """
+        Initializes the PhasicsCam object.
+
+        Sets up the instrument parameters, including exposure time, image dimensions,
+        and frame buffer size. Also initializes data streams and registers commands.
+
+        Attributes
+        ----------
+        exposure_time : int
+            The default exposure time in milliseconds.
+        image_height : int
+            The height of the images in pixels.
+        image_width : int
+            The width of the images in pixels.
+        num_frames_in_buffer : int
+            The number of frames to maintain in the data buffer.
+        images : catkit2.testbed.service.DataStream
+            A data stream to manage phase map image frames.
+        intensity : catkit2.testbed.service.DataStream
+            A data stream to manage intensity image frames.
+        """
+        super().__init__('phasics_cam')
+        # in dummy mode fake data is generated by API or device
```

Comment:
Most of the following code shouldn't be in the `__init__()`, but in the `open()`. While in the nominal case there is no difference, if there is an exception thrown, error handling is not performed in the same way. Typically, in the `__init__()` nothing should ever fail if there is a problem with the device itself.